client.go

```diff
@@ -0,0 +1,295 @@
+package socketio
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"path"
+	"strings"
+
+	"github.com/googollee/go-socket.io/engineio"
+	"github.com/googollee/go-socket.io/engineio/transport"
+	"github.com/googollee/go-socket.io/engineio/transport/polling"
+	"github.com/googollee/go-socket.io/logger"
+	"github.com/googollee/go-socket.io/parser"
+)
+
+// Server is a go-socket.io server.
+type Client struct {
+	conn      *conn
+	namespace string
+	handlers  *namespaceHandlers
+	url       string
+	opts      *engineio.Options
+}
+
+// NewServer returns a server.
+func NewClient(uri string, opts *engineio.Options) (*Client, error) {
+	// uri like http://asd.com:8080/namesapce
+
+	url, err := url.Parse(uri)
+	if err != nil {
+		return nil, err
+	}
+	namespace := url.Path
+
+	// Not allowing other than default
+	url.Path = path.Join("/socket.io", namespace)
+	url.Path = url.EscapedPath()
+	if strings.HasSuffix(url.Path, "socket.io") {
+		url.Path += "/"
+	}
+
+	client := &Client{
+		conn:      nil,
+		namespace: namespace,
+		url:       url.String(),
+		handlers:  newNamespaceHandlers(),
+		opts:      opts,
+	}
+
+	fmt.Println(client)
+
+	return client, nil
+}
+
+func (s *Client) Connect() error {
+	dialer := engineio.Dialer{
+		Transports: []transport.Transport{polling.Default},
```

Comment:
This place is written dead, the incoming opts should be read.
``` 
dialer := engineio.Dialer{
    Transports: []transport.Transport{websocket.Default},
}
if len(c.opts.Transports) > 0 {
    dialer.Transports = c.opts.Transports
}
```

Reply:
Maybe you get a better idea.

Reply:
please fix it sir